pyramid_rpc/jsonrpc.py

```diff
@@ -160,6 +160,44 @@ def __call__(self, context, request):
         return result
 
 
+class jsonrpc_batch_tween(object):
+    """ Pyramid Tween used to intercept requests and handle batch requests
+    transparently for pyramid_rpc in accordance with the JSONRPC 2.0 Specification.
+
+    """
+    def __init__(self, handler, registry):
+        self.handler = handler
+        self.registry = registry
+
+    def __call__(self, request):
+        # Check if it is a JSON request and a list.
+        try:
+            json_body = request.json_body[:]
+        except (ValueError, TypeError):
+            return self.handler(request)
```

Comment:
Frankly I would recommend splitting up this block; there's two questions at hand, is this body valid JSON and is this body an iterable. the code will be more readable with two try/except blocks.